om4labs/diags/generic_section_transport/generic_section_transport.py

```diff
@@ -0,0 +1,207 @@
+
+    # cross-check variable list with input dataset
+    complist = [
+        varname for varname in components if varname in dset_transport.variables
+    ]
+
+    # transport will eventually contain the total transport, from both
+    # u and v components if provided. It is initialized here as a
+    # list of xarray.DataArrays()
+    transport = []
+
+    # loop over transport components
+    for comp in complist:
+
+        # obtain a list of dimensions
+        component = dset_transport[comp]
+        dims = list(component.dims)
+
+        # detect x and y dimensions for this DataArray
+        xdim = [dim for dim in dims if dim.startswith("x")]
+        ydim = [dim for dim in dims if dim.startswith("y")]
+
+        if len(xdim) != 1:
+            raise RuntimeError(f"Unable to find x dimension in {comp}")
+        if len(ydim) != 1:
+            raise RuntimeError(f"Unable to find y dimension in {comp}")
+
+        xdim = xdim[0]
+        ydim = ydim[0]
+
+        # detect and sum the transport along the dimension normal to the passage definition
+        avedim = ydim if len(dset_transport[ydim]) > len(dset_transport[xdim]) else xdim
+        component = component.sum(dim=avedim)
+
+        # ensure we have a 3-d data cube at this step.
+        assert len(component.dims) == 3, f"{comp} may be missing a time coordinate"
+
+        # The data cube should also contain a singleton dimension at this step
+        component = component.squeeze()
+        assert len(component.dims) == 2, f"{comp} did not reduce to (time,depth)"
+
+        # Append this array to the total transport
+        transport.append(component)
+
+    # concatentate components along a new axis and take the sum
+    transport = xr.concat(transport, dim="component")
+    transport = transport.sum(dim="component")
+    assert (
+        len(transport.dims) == 2
+    ), f"Total transport has wrong dimensions, should be (time,depth)"
+
+    # restrict vertical range if asked
+    zcoord = transport.dims[1]
+    if zlim is not None:
+        assert isinstance(zlim, tuple), "zlim must be a tuple of depths"
+        zlim = {zcoord: slice(*zlim)}
+        transport = transport.sel(**zlim)
+
+    # sum in the verical
+    transport = transport.sum(dim=zcoord)
+
+    # check that we have a timeseries
+    assert len(transport.dims) == 1, "Dataset did not yield a timeseries"
+
+    # apply conversion factor
+    transport = transport * conversion
+
+    # create montly average if asked
+    transport = transport.resample(time="1M").mean() if monthavg else transport
+
+    # rename the output field
+    transport = transport.rename("summed transport")
+
+    # load array
+    transport.load()
+
+    return transport
+
+
+def plot(transport, ylim=None, label=None, passage_label=None, obsrange=None):
+    fig = plt.figure(figsize=(8.5, 4))
+    ax = plt.subplot(1, 1, 1)
+
+    # plot obs range
+    if obsrange is not None:
+        minval = (transport.values * 0.0) + min(obsrange)
```

Comment:
this is a bit easier to understand:

```python
minval = min(obsrange) * np.ones((len(transport.values)))

Reply:
Thanks, @raphaeldussin. I addressed this in f38464d.